src/snowflake/cli/plugins/connection/util.py

```diff
@@ -50,11 +56,44 @@ def is_regionless_redirect(conn: SnowflakeConnection) -> bool:
         *_, cursor = conn.execute_string(REGIONLESS_QUERY, cursor_class=DictCursor)
         return cursor.fetchone()["REGIONLESS"].lower() == "true"
     except:
-        # by default, assume that
         log.warning("Cannot determine regionless redirect; assuming True.")
         return True
 
 
+def guess_regioned_host_from_allowlist(conn: SnowflakeConnection) -> str | None:
+    """
+    Use SYSTEM$ALLOWLIST to find a regioned host (<account>.x.y.z.snowflakecomputing.com)
+    that corresponds to the given Snowflake connection object.
+    """
+    *_, cursor = conn.execute_string(ALLOWLIST_QUERY, cursor_class=DictCursor)
+    allowlist_tuples = json.loads(cursor.fetchone()["SYSTEM$ALLOWLIST()"])
+    for t in allowlist_tuples:
+        if t["type"] == SNOWFLAKE_DEPLOYMENT:
+            host_parts = t["host"].split(".")
```

Comment:
this can also reuse _get_region_from_host

Reply:
It can't, because we're returning the whole host, not just the region.